new help page on tiling

jo-pol opened the pull request:
Preview of the [new page](https://github.com/jo-pol/GroundForge/blob/tiling-help/docs/help/Brick-to-overlap.md)

jo-pol commented:
@d-bl/gf 
feedback on this new [help page](https://github.com/jo-pol/GroundForge/blob/tiling-help/docs/help/Brick-to-overlap.md) would be welcome

MAETempels commented:
Hallo Joke,

I miss some kind of introduction about the goal of the page.
Suggestion: *For some (older) tile definitions it may be difficult to
discover which stitch is exactly where, as these definitions divide the
ground in rectangles rather than in units of work. On this page we give a
method how to change the tiles to fit better the units of work. The new
tile definitions may have overlaps.*

*The large green tile allows you to vary stitches in two units of work. *
... *However, the second unit in this larger tile is sliced up in three
parts. *
I only see the green rectangles. I think, the large PURPLE rectangle is
meant, as this is in the picture, but not in the text.

*Any textbook will explain the bottom corners of a unit are the top corners
for the next row. Make these stitches faint with a dash.*
Previous instructions (last week) said "put a dash in the top left corner".
The line here suggests the bottom corner.

*The overlap was not implemented by design, the feature is just a "happy
accident".  *
Is this relevant for a tutorial? Might go on a "the history of GF" page :-)

*See also the tutorial
<https://github.com/jo-pol/GroundForge/blob/tiling-help/docs/help/Advanced#glue-copies-together>
pages
for more aspects of the subject. *
Which subject? (Always thought  " more aspects *on *the subject").

Op za 21 nov. 2020 om 21:43 schreef Jo Pol <notifications@github.com>:

> @d-bl/gf <https://github.com/orgs/d-bl/teams/gf>
> feedback on this new help page
> <https://github.com/jo-pol/GroundForge/blob/tiling-help/docs/help/Brick-to-overlap.md>
> would be welcome
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/d-bl/GroundForge/pull/173#issuecomment-731634017>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGYKEK7OJD4JCXZPCRWBGBLSRAQ5XANCNFSM4T563HAA>
> .
>


-- 
Groetjes van Marian

veronika commented:
I don't see the connection between "using overlap tiles" and "easily finding a particular node in a pattern".  The main problem I see is that the base tile is in the top left corner.  The way it connects to the rest of the pattern on the left and top edges is not easy to see.  You have to do mental translation to figure out how the pattern wraps around.  Also, the base tile overlaps with the footside and you can't easily see what part of the pattern nodes on the left  belong to.  

I think the problem described by Marian would be better addressed by moving the base tile into the middle of the Pattern diagram.  There should be one tile above and one tile to the left of the base.  To the right, there should be additional columns to match the edge column of the swatch size (swatch columns modulo base tile columns for simple tile, other layouts can be similarly determined).  Below it should enough rows to complete a full vertical translation of the layout (return to the same node in column 1 as the first node in the base pattern).

I don't think that changing the pattern from a brick layout to an overlap layout is a primary use case and therefore it should not be the help link from the Tile layout area.

MAETempels commented:
I thought the re-definition of the tiles was all about an easier way to find particular nodes. My confusion.

Veronika wrote: _There should be one tile above and one tile to the left of the base. To the right, there should be additional columns to match the edge column of the swatch size_ ...
I would think to the right  enough columns to complete a full horizontal translation of the layout.
Some patterns can be defined with 1 (large) column. (Or with 1 (large) row.)

So, disconnect the size of the tile definition from the parameters given at the pairs-diagram size.
Maybe ask for numbers of repeats of the tile in stead of numbers of rows/columns.

I agree with Veronika about the distinction between the left footside and the base tile. Maybe another color for the arrows?

jo-pol commented:
Renamed the page to [Change-tiles](https://github.com/jo-pol/GroundForge/blob/tiling-help/docs/help/Change-tiles.md)

The code has been tweaked for just over five years and started with some serious design flaws and implementation mistakes. We'll have to live with the bright tile in the top left corner until a complete redesign, currently I'm distracted from that task.

I want at my fingertips what I can't remember (exactly) for the same reason as having the cheat sheet permanently visible. On the user group I started with a wrong procedure, underlining how easy it is to get the procedure wrong.

I dropped the reasoning about the pros an cons. It is a subject for other pages or the user group, perhaps even the section "Larger tiles" belongs on that page. Also moved the link to the tutorial to the top. Perhaps this link should become a few concise links.

Cowboynoon commented:
Dear Jo, everyone

I started to edit the text a little to add an introduction when I realised
that the basic principles of tiling are covered in the "Creating a Pattern
Definition" tutorial and that it would be an unnecessary repeat of
information - I think your animation is wonderfully clear and should be
added to the tutorial page (inline, not as a link that opens in a new page,
if possible), and since it gives such a succinct explanation of the
buttons' functionality, I'd link to the animation alone in the "i" link,
with perhaps a link to the tutorial below the animation if anyone needs a
thorough narrative explanation.


("creating a pattern repeat" has a different title in the menu "create a
prototype" I just noticed too)

Pierre
